src/app/token.interceptor.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class TokenInterceptor implements HttpInterceptor {
+
+  constructor() {}
+
+  intercept(req: HttpRequest<any>,
+            next: HttpHandler): Observable<HttpEvent<any>> {
+
+    const userToken = localStorage.getItem('userToken');
+    if (userToken) {
```

Comment:
Just to be safe, the function should check the target of the request and only append the authorization token if our own API is the target.